File: app/src/main/java/com/college/collegeconnect/datamodels/SaveSharedPreference.java
package com.college.collegeconnect.datamodels;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

public class SaveSharedPreference {
    private static final String PREF_USER_NAME = "username";
    private static final String USER = "user";
    private static final String CHECKED_ITEM = "checked_item";
    private static final String COURSE = "course";
    private static final String BRANCH = "branch";
    private static final String SEMESTER = "semester";
    private static final String UNIT = "unit";
    private static final String CLEARALL = "clearall";
    private static final String CLEARALL1 = "clearall1";
    private static final String REF = "com.connect.collegeconnect.MyPref";
    private static final String ATTENDANCE_CRITERIA = "attendance_criteria";
    private static final String POP = "pop";
    private static final String DETAILS_UPLOADED = "uploaded";

    public static SharedPreferences getSharedPreferences(Context ctx) {
        return PreferenceManager.getDefaultSharedPreferences(ctx);
    }

    // To store upload status
    public static void setUploaded(Context ctx, Boolean var) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putBoolean(DETAILS_UPLOADED, var);
        editor.apply();
    }

    // To store email
    public static void setUserName(Context ctx, String userName) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putString(PREF_USER_NAME, userName);
        editor.apply();
    }

    // To store name
    public static void setUser(Context ctx, String user) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putString(USER, user);
        editor.apply();
    }

    //For storing theme choice
    public static void setCheckedItem(Context ctx, int path) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putInt(CHECKED_ITEM, path);
        editor.apply();
    }

    public static void setCourse(Context ctx, int num) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putInt(COURSE, num);
        editor.apply();
    }

    public static void setBranch(Context ctx, int num) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putInt(BRANCH, num);
        editor.apply();
    }

    public static void setSemester(Context ctx, int num) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putInt(SEMESTER, num);
        editor.apply();
    }

    public static void setUnit(Context ctx, int num) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putInt(UNIT, num);
        editor.apply();
    }

    public static void setClearall(Context ctx, boolean value) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putBoolean(CLEARALL, value);
        editor.apply();
    }

    public static void setClearall1(Context ctx, boolean value) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putBoolean(CLEARALL1, value);
        editor.apply();
    }

    //For on boarding screen
    public static void setRef(Context ctx, boolean value) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putBoolean(REF, value);
        editor.apply();
    }

    public static void setAttendanceCriteria(Context ctx, int attendance_criteria) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putInt(ATTENDANCE_CRITERIA, attendance_criteria);
        editor.apply();
    }

    public static void setPop(Context ctx, int pop) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putInt(POP, pop);
        editor.apply();
    }

    public static int getAttendanceCriteria(Context ctx) {
        return getSharedPreferences(ctx).getInt(ATTENDANCE_CRITERIA, 75);
    }

    public static String getUserName(Context ctx) {
        return getSharedPreferences(ctx).getString(PREF_USER_NAME, "");
    }

    public static String getUser(Context ctx) {
        return getSharedPreferences(ctx).getString(USER, "");
    }

    public static int getCheckedItem(Context ctx) {
        return getSharedPreferences(ctx).getInt(CHECKED_ITEM, 0);
    }

    public static int getCourse(Context ctx) {
        return getSharedPreferences(ctx).getInt(COURSE, 0);
    }

    public static int getBranch(Context ctx) {
        return getSharedPreferences(ctx).getInt(BRANCH, 0);
    }

    public static int getSemester(Context ctx) {
        return getSharedPreferences(ctx).getInt(SEMESTER, 0);
    }

    public static int getUnit(Context ctx) {
        return getSharedPreferences(ctx).getInt(UNIT, 0);
    }

    public static boolean getClearall(Context ctx) {
        return getSharedPreferences(ctx).getBoolean(CLEARALL, false);
    }

    public static boolean getClearall1(Context ctx) {
        return getSharedPreferences(ctx).getBoolean(CLEARALL1, false);
    }

    public static boolean getRef(Context ctx) {
        return getSharedPreferences(ctx).getBoolean(REF, false);
    }

    public static boolean getUpload(Context ctx) {
        return getSharedPreferences(ctx).getBoolean(DETAILS_UPLOADED, false);
    }

    public static int getPop(Context ctx) {
        return getSharedPreferences(ctx).getInt(POP, 1);
    }

    //Clear data on logout
    public static void clearUserName(Context ctx) {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.remove(PREF_USER_NAME);
        editor.remove(USER);
        editor.remove(CHECKED_ITEM);
        editor.remove(COURSE);
        editor.remove(BRANCH);
        editor.remove(SEMESTER);
        editor.remove(CLEARALL1);
        editor.remove(CLEARALL);
        editor.remove(UNIT);
        editor.remove(ATTENDANCE_CRITERIA);
        editor.apply();
    }

}

File: app/src/main/java/com/college/collegeconnect/datamodels/SaveSharedPreference.kt
package com.college.collegeconnect.datamodels

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

object SaveSharedPreference {
    private const val PREF_USER_NAME = "username"
    private const val USER = "user"
    private const val CHECKED_ITEM = "checked_item"
    private const val COURSE = "course"
    private const val BRANCH = "branch"
    private const val SEMESTER = "semester"
    private const val UNIT = "unit"
    private const val CLEARALL = "clearall"
    private const val CLEARALL1 = "clearall1"
    private const val REF = "com.connect.collegeconnect.MyPref"
    private const val ATTENDANCE_CRITERIA = "attendance_criteria"
    private const val POP = "pop"
    private const val DETAILS_UPLOADED = "uploaded"
    private const val COLLEGE = "college"
    private const val REVIEW = "review"
    private fun getSharedPreferences(ctx: Context?): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(ctx)
    }

    // To store upload status
    @JvmStatic
    fun setUploaded(ctx: Context?, `var`: Boolean?) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putBoolean(DETAILS_UPLOADED, `var`!!)
        editor.apply()
    }

    // To store email
    @JvmStatic
    fun setUserName(ctx: Context?, userName: String?) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putString(PREF_USER_NAME, userName)
        editor.apply()
    }

    // To store name
    @JvmStatic
    fun setUser(ctx: Context?, user: String?) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putString(USER, user)
        editor.apply()
    }

    //For storing theme choice
    @JvmStatic
    fun setCheckedItem(ctx: Context?, path: Int) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putInt(CHECKED_ITEM, path)
        editor.apply()
    }

    @JvmStatic
    fun setCourse(ctx: Context?, num: Int) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putInt(COURSE, num)
        editor.apply()
    }

    @JvmStatic
    fun setBranch(ctx: Context?, num: Int) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putInt(BRANCH, num)
        editor.apply()
    }

    @JvmStatic
    fun setSemester(ctx: Context?, num: Int) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putInt(SEMESTER, num)
        editor.apply()
    }

    @JvmStatic
    fun setUnit(ctx: Context?, num: Int) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putInt(UNIT, num)
        editor.apply()
    }

    @JvmStatic
    fun setClearall(ctx: Context?, value: Boolean) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putBoolean(CLEARALL, value)
        editor.apply()
    }

    @JvmStatic
    fun setClearall1(ctx: Context?, value: Boolean) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putBoolean(CLEARALL1, value)
        editor.apply()
    }

    //For on boarding screen
    @JvmStatic
    fun setRef(ctx: Context?, value: Boolean) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putBoolean(REF, value)
        editor.apply()
    }

    @JvmStatic
    fun setAttendanceCriteria(ctx: Context?, attendance_criteria: Int) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putInt(ATTENDANCE_CRITERIA, attendance_criteria)
        editor.apply()
    }

    @JvmStatic
    fun setPop(ctx: Context?, pop: Int) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putInt(POP, pop)
        editor.apply()
    }

    @JvmStatic
    fun setCollege(ctx: Context?, college: String?) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putString(COLLEGE, college)
        editor.apply()
    }

    @JvmStatic
    fun setRev(ctx: Context?, value: Boolean) {
        val editor = getSharedPreferences(ctx).edit()
        editor.putBoolean(REVIEW, value)
        editor.apply()
    }

    fun getAttendanceCriteria(ctx: Context?): Int {
        return getSharedPreferences(ctx).getInt(ATTENDANCE_CRITERIA, 75)
    }

    @JvmStatic
    fun getUserName(ctx: Context?): String? {
        return getSharedPreferences(ctx).getString(PREF_USER_NAME, "")
    }

    @JvmStatic
    fun getUser(ctx: Context?): String? {
        return getSharedPreferences(ctx).getString(USER, "")
    }

    @JvmStatic
    fun getCheckedItem(ctx: Context?): Int {
        return getSharedPreferences(ctx).getInt(CHECKED_ITEM, 2)
    }

    @JvmStatic
    fun getCourse(ctx: Context?): Int {
        return getSharedPreferences(ctx).getInt(COURSE, 0)
    }

    @JvmStatic
    fun getBranch(ctx: Context?): Int {
        return getSharedPreferences(ctx).getInt(BRANCH, 0)
    }

    @JvmStatic
    fun getSemester(ctx: Context?): Int {
        return getSharedPreferences(ctx).getInt(SEMESTER, 0)
    }

    @JvmStatic
    fun getUnit(ctx: Context?): Int {
        return getSharedPreferences(ctx).getInt(UNIT, 0)
    }

    @JvmStatic
    fun getClearall(ctx: Context?): Boolean {
        return getSharedPreferences(ctx).getBoolean(CLEARALL, false)
    }

    fun getClearall1(ctx: Context?): Boolean {
        return getSharedPreferences(ctx).getBoolean(CLEARALL1, false)
    }

    @JvmStatic
    fun getRef(ctx: Context?): Boolean {
        return getSharedPreferences(ctx).getBoolean(REF, false)
    }

    @JvmStatic
    fun getUpload(ctx: Context?): Boolean {
        return getSharedPreferences(ctx).getBoolean(DETAILS_UPLOADED, false)
    }

    @JvmStatic
    fun getPop(ctx: Context?): Int {
        return getSharedPreferences(ctx).getInt(POP, 1)
    }

    @JvmStatic
    fun getCollege(ctx: Context?): String? {
        return getSharedPreferences(ctx).getString(COLLEGE, "other")
    }

    @JvmStatic
    fun getReview(ctx: Context?): Boolean {
        return getSharedPreferences(ctx).getBoolean(REVIEW, false)
    }

    //Clear data on logout
    @JvmStatic
    fun clearUserName(ctx: Context?) {
        val editor = getSharedPreferences(ctx).edit()
        editor.remove(PREF_USER_NAME)
        editor.remove(USER)
        editor.remove(CHECKED_ITEM)
        editor.remove(COURSE)
        editor.remove(BRANCH)
        editor.remove(SEMESTER)
        editor.remove(CLEARALL1)
        editor.remove(CLEARALL)
        editor.remove(UNIT)
        editor.remove(ATTENDANCE_CRITERIA)
        editor.remove(REF)
        editor.remove(COLLEGE)
        editor.apply()
    }
}